main.c

```diff
@@ -99,15 +99,19 @@ void reset_mp(void) {
     mp_obj_list_init(mp_sys_argv, 0);
 }
 
-bool maybe_run(const char* filename, pyexec_result_t* exec_result) {
-    mp_import_stat_t stat = mp_import_stat(filename);
-    if (stat != MP_IMPORT_STAT_FILE) {
-        return false;
+bool maybe_run_list(const char ** filenames, pyexec_result_t* exec_result) {
+
+    for (int i = 0; *(filenames[i]); i++) {
```

Comment:
Implicit C int -> bool confuses my C++ and python brain. Mind making this `filenames[i] != ""`?

Reply:
I agree, that would be easier to read.